addons/payment_mollie/models/payment_transaction.py

```diff
@@ -76,6 +77,55 @@ def _mollie_prepare_payment_request_payload(self):
             # redirection, we include it in the redirect URL to be able to match the transaction.
             'redirectUrl': f'{redirect_url}?ref={self.reference}',
             'webhookUrl': f'{webhook_url}?ref={self.reference}',
+            'billingAddress': self._prepare_billing_address(),
+            'lines': [{
+                "description": 'Odoo purchase',
+                "quantity": 1,
+                "unitPrice": {
+                    "currency": self.currency_id.name,
+                    "value": f"{self.amount:.{decimal_places}f}"
+                },
+                "totalAmount": {
+                    "currency": self.currency_id.name,
+                    "value": f"{self.amount:.{decimal_places}f}"
+                },
+            }],
+        }
+
+    def _prepare_billing_address(self):
+        given_name, family_name = payment_utils.split_partner_name(self.partner_name)
+        missing_fields = []
+
+        if not given_name:
+            missing_fields.append(_("Given name"))
+        if not family_name:
+            missing_fields.append(_("Family name"))
```

Comment:
I think First and Last name are more universal
```suggestion
            missing_fields.append(_("First name"))
        if not family_name:
            missing_fields.append(_("Last name"))
```